asserts/database.go

```diff
@@ -45,14 +45,7 @@ func (e *NotFoundError) Error() string {
 }
 
 func (e *NotFoundError) Is(err error) bool {
-	// TODO: replace IsNotFound usages for errors.Is(err, &NotFoundError{})
-	return IsNotFound(err)
-}
-
-// IsNotFound returns whether err is an assertion not found error.
-func IsNotFound(err error) bool {
-	_, ok := err.(*NotFoundError)
-	return ok
+	return errors.Is(err, &NotFoundError{})
```

Comment:
This creates a recursion loop which causes a stack overflow panic

Reply:
Thank you for pointing out the [HACKING.md](https://github.com/snapcore/snapd/blob/master/HACKING.md) document. I am running Fedora, so I was having trouble with some dependencies.
I fixed the issue, installed an Ubuntu VM and ran the unit tests there instead.

Reply:
I noticed that it was advised in the [HACKING.md](https://github.com/snapcore/snapd/blob/master/HACKING.md#guidelines-for-contributors) document to not force push.
Please let me know if I need to squash my commits before merging.

Reply:
Yes, if you could squash before merging (after the PR has 2 approvals), that would be great. Thanks